constraint-solver/src/solver.rs

```diff
@@ -156,6 +171,17 @@ impl<T: FieldElement, V: Ord + Clone + Hash + Eq + Display + Debug> Solver<T, V>
         !equivalences.is_empty()
     }
 
+    fn solve_with_backtracking(&mut self) -> Result<bool, Error> {
+        let assignments = Backtracker::new(self).get_unique_assignments()?;
```

Comment:
This could also return a vector of effects. Do you think it makes sense to unify the interface of the sub-solvers?

Reply:
Then we could also sort them by priorities. One thing to note is that we have some sub-solvers that are run on individual items and then can be triggered if a variable in that item was updated and some sub-solvers that are run on the whole system...

Reply:
Yes, should be another PR, we should just do this before we start adding more and more sub-solvers.